src/charm.py

```diff
@@ -120,15 +142,33 @@ def _login_ui_layer(self) -> Layer:
         return Layer(pebble_layer)
 
     def _get_kratos_endpoint_info(self) -> Optional[str]:
-        kratos_public_url = ""
         if self.model.relations[self._kratos_relation_name]:
             try:
                 kratos_endpoints = self.kratos_endpoints.get_kratos_endpoints()
                 kratos_public_url = kratos_endpoints["public_endpoint"]
+                return kratos_public_url
             except KratosEndpointsRelationDataMissingError:
                 logger.info("No kratos-endpoint-info relation data found")
+                return None
+        return None
+
+    def _update_login_ui_endpoint_relation_data(self, event: RelationEvent) -> None:
+        endpoint = self.ingress.url if self.ingress.is_ready() else ""
 
-        return kratos_public_url
+        logger.info(f"Sending login ui endpoint info: endpoint - {endpoint[0]}")
+
+        self.endpoint_provider.send_endpoints_relation_data(endpoint[0])
```

Comment:
`endpoint_provider` is not declared, which causes the charm to fail when this method is run

Reply:
thanks for the catch, fixed now